src/main/java/spoon/reflect/path/impl/CtPathImpl.java

```diff
@@ -33,9 +33,70 @@ public <T extends CtElement> List<T> evaluateOn(CtElement... startNode) {
 		for (CtPathElement element : elements) {
 			filtered = element.getElements(filtered);
 		}
+		if (filtered.isEmpty()) {
+			List<String> cls_name_list = new LinkedList<>();
+			CtType<?> ctType = null;
+			for (CtPathElement element : elements) {
```

Comment:
I am not very well versed with the CtPath API, but this seems like it re-implements existing functionality and we should instead adjust the matching logic of the path elements?

Reply:
Seeing issue #4984 , yeah I want to reuse the case that with no parameter for matching shadow elements.
The matching logic do differ, so it may also be a good idea to extract a new method `evaluateOnShadowModel` as you suggested.